server/base/src/main/java/org/apache/accumulo/server/compaction/FileCompactor.java

```diff
@@ -517,8 +518,10 @@ private void compactLocalityGroup(String lgName, Set<ByteSequence> columnFamilie
         majCStats.add(lgMajcStats);
         writeSpan.end();
       }
-
     } catch (Exception e) {
+      if (e instanceof InterruptedIOException) {
+        throw new CompactionCanceledException();
```

Comment:
Nice to include the cause

```suggestion
        throw new CompactionCanceledException(e);
```